### ISSUE TITLE
Default Property Values not Being Set in v0.5.0

### ISSUE THREAD

**Issue by colemanator** [rexlabsio/data-transfer-object#24]
Defaults defined in the static `getDefaults` method and not being used when the DTO is created in v0.5.0.

## Detailed description
Defaults defined in the static `getDefaults` method and not being used because the following method in `src/Factory.php` references a undefined variable: `$defaults`.

```php
/**
     * @param ClassData $classData
     * @param array $useStatements
     * @return Property[]
     */
    public function mapClassToPropertyTypes(ClassData $classData, array $useStatements): array
    {
        return $this->mapAssoc(
            function (string $docType, string $name) use ($classData, $useStatements): Property {
                $types = $this->mapTypes(
                    $classData->namespace,
                    $useStatements,
                    explode('|', $docType)
                );

                $arrayTypes = $this->mapArrayTypes($types);

                return new Property(
                    $this,
                    $name,
                    $types,
                    $arrayTypes,
                    array_key_exists($name, $classData->defaults),
                    $defaults[$name] ?? null
                );
            },
            $this->extractDocPropertyTypes($classData->docComment)
        );
    }
```

## Context
Looking to use DTOs for constructing events for analytics, I can probably work around it for now with a hack. I could update to `1.0.0` but that would require more work (due to breaking changes) and I would have to change the projects min stability. 

## Possible implementation
Can we create a patch `0.5.1` and Change this line `$defaults[$name] ?? null` to `$classData->defaults[$name] ?? null`. Happy to make the PR.

## Your environment

* PHP 7.3.23
* Linux, Alpine 3.12
* https://github.com/rexlabsio/spoke/pull/907

**Comment by colemanator** [comment]
This bug also bypasses the check for uninitialised properties as all properties default to `null` if not set.

**Comment by ababkov** [comment]
How easy / hard would it be for you guys to upgrade to the v1 rc?

**Comment by colemanator** [comment]
I think the largest change would be I'd have to make sure we are using the correct flags as they changed. Not sure if there will be able conflicts when installing 🤷‍♂️ 

It's less work to just fix the bug, it's a one line change and a single git tag. I mean it's not blocking anything I just overloaded the constructor. If we don't want to fix the bug and instead just require consumers update a major version that's fine as well.

**Comment by lachlankrautz** [comment]
Fixed in 0.5.2
